mdtraj/geometry/alignment.py

```diff
@@ -269,3 +269,33 @@ def rmsd_qcp(conformation1, conformation2):
     max_eigenvalue = scipy.optimize.newton(f, E0, df)
     rmsd = np.sqrt(np.abs(2.0 * (E0 - max_eigenvalue) / n_atoms))
     return rmsd
+
+def compute_average_structure(xyz):
+    """Compute the average structure from a set of frames.
+
+    The frames are first aligned to minimize the RMSD from the average structure,
+    and then the average is returned.
+
+    Parameters
+    ----------
+    xyz : ndarray, shape = (n_frames, n_atoms, 3)
+        xyz coordinates of each atom in each frame
+
+    Returns
+    -------
+    average : ndarray, shape = (n_atoms, 3)
+        the average structure
+    """
+    n_frames = xyz.shape[0]
+    n_atoms = xyz.shape[1]
+    candidate = xyz[0]
+    
+    # In practice there is usually negligible improvement after the second iteration.
+    # We do three just to be safe.
+    
+    for iteration in range(3):
+        average = np.zeros((n_atoms, 3))
+        for frame in range(n_frames):
+            average += transform(xyz[frame], candidate)
```

Comment:
Are you just aligning to the first frame? Is this approach equivalent to selecting the reference structure for alignment that minimizes the RMSD, as the docstring suggests?